content/en/docs/instrumentation/java/automatic/annotations.md

```diff
@@ -55,6 +55,22 @@ denotes its duration and provides any thrown exceptions. By default, the span
 name will be `<className>.<methodName>`, unless a name is provided as an
 argument to the annotation.
 
+If the return type of the method annotated by `@WithSpan` is one of the below "future promise" types,
```

Comment:
In terms of an alternative formulation for "future promise" (and a slight reorg of the phrase), how about this:

> ... is one of the [Future]-like types listed below, then ... until the future completes:

[Future]: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html

(I don't know if you'll want to link to https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/util/concurrent/Future.html or https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Future.html or something else.)

Future-like is used here to mean: a type that represents an asynchronous computation and possible future result.

Reply:
I like that, especially since we don't actually support `java.util.concurrent.Future` 😄